### ISSUE TITLE
LIF model error

### ISSUE THREAD

**Issue by kupiqu** [DynaSim/DynaSim#454]
I'm testing dsDemos (dev branch) in the upcoming Octave version 4.4. There is an important error in the integrate-and-fire model. It took me a while to see what was going on, why it failed in Octave and not in Matlab. My conclusion is that there is a bug in Dynasim, which is hidden by ~Matlab stupidity~ a Matlab bug. Let me explain...

There are certainly less spikes than timesteps and the variables storing spikes information do so in 0/1 values, but these variables are not initialized in the solve file, so their final temporal dimension is limited to the last spike. And if there are no spikes then the variable is not even created, but dsSimulate tries to get it as an output argument.

This issue properly fails in Octave, but Matlab just "fixes" this. Don't think this is sth good for a moment, as the developer might want sth entirely different. Why is Matlab assuming a variable is of a certain size if it doesn't even exist!?! Or if it does exist, why Matlab is changing its size! That's a crazy bug in Matlab, that in this case hid the bug in the solve file. And that Matlab bug may be addressed (fixed) at some point or "work" differently in different Matlab versions.

@jsherfey could you please try to correct this in the creation of solve files that use conditionals (**i.e. those variables should be initialized to their expected size**), such as in the LIF model?

I also saw that rng_wrapper was called twice in the LIF solve file. Not sure this happens in all solve files or sth weird is going on for the LIF model... Please try to take a look at this as well, if you can.

**Comment by kupiqu** [comment]
This is the piece of code:

```
%% Networks of integrate-and-fire neurons with axonal delays and refractory period
% tau [ms]: membrane time constant (RC)
% tref [ms]: absolute refractory period
% delay [ms]: axonal delay
% tspike: reserved variable for storing past spike times when monitoring spikes

% 1 E-cell driving 1 I-cell
LIF={
    'dV/dt=(E-V+R*I+noise*randn-@isyn)/tau; V(0)=-65'
    'if(any(t<tspike+tref,1))(V=reset)'
    'tau=10; tref=10; E=-70; thresh=-55; reset=-75; R=9; I=1.55; noise=100'
    'monitor V.spikes(thresh)'
     };

iampa={
  'gAMPA=.5; EAMPA=0; tauD=2; tauR=0.4; delay=15'
  'f(x) = (exp(-x/tauD)-exp(-x/tauR)).*(x>0)'
  'Isyn(V) = gAMPA.*sum(f(t-tspike_pre-delay)).*(V-EAMPA)'
  '@isyn += Isyn(V_post)'
};

s=[];
s.populations(1).name='E';
s.populations(1).equations=LIF;
s.populations(2).name='I';
s.populations(2).equations=LIF;
s.populations(2).parameters={'I',0,'noise',0};
s.connections(1).direction='E->I';
s.connections(1).mechanism_list='iampa';
s.mechanisms(1).name='iampa';
s.mechanisms(1).equations=iampa;

data=dsSimulate(s,'time_limits',[0 200],'solver','rk1','dt',.01);
```

**Comment by kupiqu** [comment]
Fixed with merge #503
